src/Symfony/Component/String/Inflector/FrenchInflector.php

```diff
@@ -0,0 +1,257 @@
+        // Les mots finissant par "ou" prennent un "s" sauf bijou, caillou, chou, genou, hibou, joujou, pou
+        ['/(bij|caill|ch|gen|hib|jouj|p)oux$/i', '\1ou'],
+
+        // French titles
+        ['/^mes(dame|demoiselle)s$/', 'ma\1'],
+        ['/^Mes(dame|demoiselle)s$/', 'Ma\1'],
+        ['/^mes(sieur|seigneur)s$/', 'mon\1'],
+        ['/^Mes(sieur|seigneur)s$/', 'Mon\1'],
+
+        //Default rule
+        ['/s$/i', ''],
+    ];
+
+    /**
+     * A list of words which should not be inflected.
+     * This list is only used by singularize.
+     */
+    private static $uninflected = [
+        //Liste des mots qui finissent par un s
+        'abcès',
+        'accès',
+        'abus',
+        'albatros',
+        'anchois',
+        'anglais',
+        'autobus',
+        'brebis',
+        'carquois',
+        'cas',
+        'chas',
+        'colis',
+        'concours',
+        'corps',
+        'cours',
+        'cyprès',
+        'décès',
+        'devis',
+        'discours',
+        'dos',
+        'embarras',
+        'engrais',
+        'entrelacs',
+        'excès',
+        'fois',
+//        'fonds',
+        'gâchis',
+        'gars',
+        'glas',
+//        'guet-apens',
+        'héros',
+        'intrus',
+        'jars',
+        'jus',
+        'kermès',
+        'lacis',
+        'legs',
+        'lilas',
+        'marais',
+        'mars',
+        'matelas',
+        'mépris',
+        'mets',
+        'mois',
+        'mors',
+        'obus',
+        'os',
+        'palais',
+        'paradis',
+        'parcours',
+        'pardessus',
+        'pays',
+        'plusieurs',
+        'poids',
+        'pois',
+        'pouls',
+        'printemps',
+        'processus',
+        'progrès',
+        'puits',
+        'pus',
+        'rabais',
+        'radis',
+        'recors',
+        'recours',
+        'refus',
+        'relais',
+        'remords',
+        'remous',
+        'rhinocéros',
+        'repas',
+        'rubis',
+        'sas',
+        'secours',
+        'souris',
+        'succès',
+        'talus',
+        'tapis',
+        'taudis',
+        'temps',
+        'tiers',
+        'univers',
+        'velours',
+        'verglas',
+        'vernis',
+        'virus',
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function singularize(string $plural): array
+    {
+        //Inflected words
+        if ($this->isInflectedWord($plural)) {
+            return [$plural];
+        }
+
+        //Exceptions rules
+        foreach (self::$singularizeRegexp as $rule) {
+            list($regexp, $replace) = $rule;
+
+            if (1 === preg_match($regexp, $plural)) {
+                return [preg_replace($regexp, $replace, $plural)];
+            }
+        }
+
+        // Default rule
+        return [$plural];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function pluralize(string $singular): array
+    {
+        //Inflected words
```

Comment:
I think all inline comments can be removed, the code is self-explanatory (and always accurate)